gateway/webhook/webhook.go

```diff
@@ -179,26 +177,33 @@ func (webhook *HandleT) RequestHandler(w http.ResponseWriter, r *http.Request) {
 		r.Header.Set("Content-Type", "application/json")
 	}
 
-	done := make(chan webhookErrorRespT)
-	req := webhookT{request: r, writer: &w, done: done, sourceType: sourceDefName, writeKey: writeKey}
+	done := make(chan transformerResponse)
+	req := webhookT{request: r, writer: w, done: done, sourceType: sourceDefName, writeKey: writeKey}
 	webhook.requestQ[sourceDefName] <- &req
 
 	// Wait for batcher process to be done
 	resp := <-done
 	webhook.gwHandle.IncrementAckCount(1)
 	atomic.AddUint64(&webhook.ackCount, 1)
-	webhook.gwHandle.TrackRequestMetrics(resp.err)
-	if resp.err != "" {
+	webhook.gwHandle.TrackRequestMetrics(resp.Err)
+
+	if resp.Err != "" {
 		code := http.StatusBadRequest
-		if resp.statusCode != 0 {
-			code = resp.statusCode
+		if resp.StatusCode != 0 {
+			code = resp.StatusCode
 		}
-		pkgLogger.Infof("IP: %s -- %s -- Response: %d, %s", misc.GetIPFromReq(r), r.URL.Path, code, resp.err)
-		http.Error(w, resp.err, code)
-	} else {
-		pkgLogger.Debugf("IP: %s -- %s -- Response: 200, %s", misc.GetIPFromReq(r), r.URL.Path, response.GetStatus(response.Ok))
-		_, _ = w.Write([]byte(response.GetStatus(response.Ok)))
+		pkgLogger.Infof("IP: %s -- %s -- Response: %d, %s", misc.GetIPFromReq(r), r.URL.Path, code, resp.Err)
+		http.Error(w, resp.Err, code)
+		return
 	}
+
+	payload := []byte(response.Ok)
+	if resp.OutputToSource != nil {
```

Comment:
My understanding is that this is test-specific code that shouldn't be needed. Can't we simply mock the http server and send the responses we want through this channel only?
Adding a separate test-only branch here has as an effect to cover the test-only branch with tests, leaving the actual code that runs in the actual system uncovered.